lib/rubocop/cop/rails/top_level_include.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks that `include`, `extend` and `prepend` exists at
+      # the top level.
+      # Using these at the top level affects the behavior of `Object`.
+      # Under the `app` directory of the Rails application, the convention
+      # that a file name and a class name are mapped is applied. There will
+      # not be using `include`, `extend` and `prepend` at the top level.
+      # Let's use it inside `class` or `module`.
+      #
+      # @example
+      #   # bad
+      #   include M
+      #
+      #   class C
+      #   end
+      #
+      #   # bad
+      #   extend M
+      #
+      #   class C
+      #   end
+      #
+      #   # bad
+      #   prepend M
+      #
+      #   class C
+      #   end
+      #
+      #   # good
+      #   class C
+      #     include M
+      #   end
+      #
+      #   # good
+      #   class C
+      #     extend M
+      #   end
+      #
+      #   # good
+      #   class C
+      #     prepend M
+      #   end
+      class TopLevelInclude < Cop
+        MSG = '`%s` is used at the top level. Use inside `class` or `module`.'
+              .freeze
+
+        def_node_matcher :include_statement, <<-PATTERN
+          (send nil? ${:include :extend :prepend}
+            (const nil? _))
+        PATTERN
+
+        def on_send(node)
+          return unless (statement = include_statement(node))
```

Comment:
Hm. Looks like you managed to trick `Lint/AssignmentInCondition` by using parentheses. 😅 Will open an issue for that.

Reply:
Never mind. This is the intended behaviour. Can be disabled by setting `AllowSafeAssignment: false`. 🙂